### PR TITLE
rss_aggregator: Limit item description to 500 characters

### PR DESCRIPTION
**Pull request by mtrojan-ub**
... not sure if it is necessary to improve the algorithm (e.g. skip entities and multibyte characters)

### CODE REVIEW

File: cpp/lib/include/HtmlUtil.h [hunk 1]
```diff
@@ -107,4 +107,8 @@ size_t ExtractAllLinks(const std::string &html_document, std::vector<std::string
 std::string StripHtmlTags(const std::string &text_with_optional_tags, const bool replace_entities = true);
 
 
+/** \brief Shorten a text but keep HTML structure. */
+std::string ShortenText(const std::string &html_document, const unsigned max_length);
```
Comment (ruschein) [inline cpp/lib/include/HtmlUtil.h:111]: size_t would be better because that's what std::string::size returns.

File: cpp/lib/src/HtmlUtil.cc [hunk 2]
```diff
@@ -1019,4 +1019,32 @@ std::string StripHtmlTags(const std::string &text_with_optional_tags, const bool
...
+std::string ShortenText(const std::string &html_text, const unsigned max_length) {
+    std::string shortened_text;
+    shortened_text.reserve(html_text.size());
+
+    bool in_tag(false);
+    unsigned length(0);
+
+    static const std::string placeholder("...");
+    unsigned max_length_without_placeholder(max_length - placeholder.length());
+    for (auto ch(html_text.cbegin()); ch != html_text.cend(); ++ch) {
```
Comment (ruschein) [inline cpp/lib/src/HtmlUtil.cc:1031]: Any particular reason why you didn't use an implicit for-loop?

**Review by ruschein — CHANGES_REQUESTED** [review]

File: cpp/lib/src/HtmlUtil.cc (hunk 2 above)
Comment (mtrojan-ub) [reply cpp/lib/src/HtmlUtil.cc:1031]: I tried to find an example for a similar implementation and copied code from HtmlUtil::StripHtmlTags.

File: cpp/lib/include/HtmlUtil.h (hunk 1 above)
Comment (mtrojan-ub) [reply cpp/lib/include/HtmlUtil.h]: OK, i changed it (and the length type inside the function as well)

File: cpp/lib/src/HtmlUtil.cc (hunk 2 above)
Comment (ruschein) [reply cpp/lib/src/HtmlUtil.cc:1031]: That is ancient code.  Please use "for (const char ch : html_text)" and replace "*ch" with "ch" everywhere in the loop body.

**Pull request merged**